theine/theine.py

```diff
@@ -8,53 +8,54 @@
 from threading import Event, Thread
 from typing import (
     Any,
-    Callable,
+    Awaitable, Callable,
     Dict,
     Hashable,
     List,
     Optional,
-    Tuple,
+    TYPE_CHECKING, Tuple,
     Type,
     TypeVar,
     Union,
-    cast,
-)
+    cast, )
 
+from mypy_extensions import KwArg, VarArg
 from theine_core import ClockProCore, LruCore, TlfuCore
 from typing_extensions import ParamSpec, Protocol
 
 from theine.exceptions import InvalidTTL
 from theine.models import CacheStats
 
+P = ParamSpec("P")
+R = TypeVar("R", covariant=True, bound=Any)
+R_A = TypeVar("R_A", covariant=True, bound=Union[Awaitable[Any], Callable[..., Any]])
```

Comment:
seems R is replaced by R_A? but R_A bound to `Union[Awaitable[Any], Callable[..., Any]]`, so this code
```python
@Memoize(Cache("tlfu", 1000), None)
def foo(id: int, m: Mock) -> Dict[str, int]:
    m(id)
    return {"id": id}
```
will not pass mypy check. I already tried and this is the error:
```
error: Value of type variable "R_A" of "__call__" of "Memoize" cannot be "dict[str, int]"  [type-var]
```